lib/URL-impl.js

```diff
@@ -19,9 +20,12 @@ exports.implementation = class URLImpl {
       throw new TypeError("Invalid URL");
     }
 
+    const query = parsedURL.query || "";
+
     this._url = parsedURL;
 
-    // TODO: query stuff
+    this._query = URLSearchParams.createImpl([query], { doNotStripQMark: true });
```

Comment:
`createImpl` calls the `URLSearchParams` [constructor](https://url.spec.whatwg.org/#dom-urlsearchparams-urlsearchparams), while the `URL` constructor instead calls the [new `URLSearchParams` object](https://url.spec.whatwg.org/#concept-urlsearchparams-new) algorithm. The difference is that the former strips the leading question mark (if any), while the latter does not.